src/enlyze/client.py

```diff
@@ -150,6 +183,88 @@ def get_variables(
             for variable in self._get_variables(machine.uuid)
         ]
 
+    def _get_paginated_timeseries(
+        self,
+        *,
+        machine_uuid: str,
+        start: datetime,
+        end: datetime,
+        variables: Sequence[str],
+        resampling_interval: Optional[int],
+    ) -> Iterator[timeseries_api_models.TimeseriesData]:
+        params: dict[str, Any] = {
+            "appliance": machine_uuid,
+            "start_datetime": start.isoformat(),
+            "end_datetime": end.isoformat(),
+            "variables": ",".join(variables),
+        }
+
+        if resampling_interval:
+            params["resampling_interval"] = resampling_interval
+
+        return self._timeseries_api_client.get_paginated(
+            "timeseries", timeseries_api_models.TimeseriesData, params=params
+        )
+
+    def _get_timeseries(
+        self,
+        start: datetime,
+        end: datetime,
+        variables: Union[
+            Sequence[user_models.Variable],
+            Mapping[user_models.Variable, user_models.ResamplingMethod],
+        ],
+        resampling_interval: Optional[int] = None,
+    ) -> Optional[user_models.TimeseriesData]:
+        try:
+            variables_sequence, variables_query_parameter_list = (
+                _get_variables_sequence_and_query_parameter_list(
+                    variables, resampling_interval
+                )
+            )
+        except ValueError as e:
+            raise EnlyzeError from e
+
+        start, end, machine_uuid = validate_timeseries_arguments(
+            start, end, variables_sequence
+        )
+
+        try:
+            chunks = chunk(
+                variables_query_parameter_list,
+                MAXIMUM_NUMBER_OF_VARIABLES_PER_TIMESERIES_REQUEST,
+            )
+        except ValueError as e:
+            raise EnlyzeError from e
+
+        chunks_pages = (
+            self._get_paginated_timeseries(
+                machine_uuid=machine_uuid,
+                start=start,
+                end=end,
+                variables=chunk,
+                resampling_interval=resampling_interval,
+            )
+            for chunk in chunks
+        )
+
+        timeseries_data_chunked = [
+            _get_timeseries_data_from_pages(pages) for pages in chunks_pages
+        ]
```

Comment:
This will materialize all the responses already, so merging them in a later step will result in twice the memory footprint. Did you consider looping over the chunked pages while constructing the final object step by step (get pages first chunk, merge, drop response, get pages of next chunk, merge, drop response, ....)?

Reply:
There's no limit imposed by the SDK w.r.t. the amount of data we're pulling from the API, so it can easily be hundreds of megabytes when fetching longer time frames. We shouldn't be wasteful with memory when we already know that we're dealing with a lot of data and it's easy to fix.

FYI: we already use the SDK in our own services, so it has a direct impact on our cluster as well.

Reply:
> drop response

How do you expect this to happen (implementation wise) in your proposed solution?